### DISCUSSION TITLE
export 'fabric' (imported as 'fabric') was not found in 'fabric'

### DISCUSSION THREAD

**Discussion by gsuherga** [fabricjs/fabric.js#10145] [Q&A] [answered]
Hello there,

I am trying to use the library in a personal project. I am using it on a IIFE function that works as I have been using it without fabric before. 

I have installed it using npm install fabric --save and then trying to import it in my javascript function with:

import { fabric } from "fabric";

but I got the message that was not found:

export 'fabric' (imported as 'fabric') was not found in 'fabric' (possible exports: ActiveSelection, BaseBrush, BaseFabricObject, Canvas, Canvas2dFilterBackend, CanvasDOMManager, Circle, CircleBrush, ClipPathLayout, Color, Control, Ellipse, FabricImage, FabricObject, FabricText, FitContentLayout, FixedLayout, Gradient, Group, IText, Image, InteractiveFabricObject, Intersection, LayoutManager, LayoutStrategy, Line, Object, Observable, Path, Pattern, PatternBrush, PencilBrush, Point, Polygon, Polyline, Rect, Shadow, SprayBrush, StaticCanvas, StaticCanvasDOMManager, Text, Textbox, Triangle, WebGLFilterBackend, cache, classRegistry, config, controlsUtils, createCollectionMixin, filters, getCSSRules, getEnv, getFabricDocument, getFabricWindow, getFilterBackend, iMatrix, initFilterBackend, isPutImageFaster, isWebGLPipelineState, loadSVGFromString, loadSVGFromURL, parseAttributes, parseFontDeclaration, parsePointsAttribute, parseSVGDocument, parseStyleAttribute, parseTransformAttribute, runningAnimations, setEnv, setFilterBackend, util, version)

I have been trying to find help about this and have not found anything. Please, I am pretty new on this, any help is welcome. 

The code looks like this:

```
import { fabric } from "fabric";

(function () {
    
    var canvas = new fabric.Canvas('c3');
    canvas.add(new fabric.Circle({radius: 30, fill: '#f55', top: 100, left: 100}));

    canvas.item(0).set({
        borderColor: 'red',
        cornerColor: 'green',
        cornerSize: 6,
        transparentCorners: false
    });
    canvas.setActiveObject(canvas.item(0));
    this.__canvases.push(canvas);
})();
```
Thanks, 

Jesús

**Comment by asturur** [comment, marked as answer]
please read this
https://fabricjs.github.io/docs/upgrading/upgrading-to-fabric-60/

**Reply by gsuherga** [reply]
Yes, that was the reason. I was trying to use the syntax of the old version with the new version. 

I will read all the new documentation. There are two websites, aren't there?

Thanks.

**Reply by asturur** [reply]
The new one should replace fabricjs.com but i need to fix https first and improve the docs a bit more at least to replace it
And i don't have time.
Maybe i should just do it and lessen the confusion, it will never be perfect anyway

**Reply by Itradetime** [reply]
Hey @asturur, I am a beginner and was looking for some example code snippets as I am starting a new fabric project. I noticed that the [docs](https://fabricjs.com/docs/) link points to an older version, which can be a bit confusing for noobs(like me).

Also, some of the links in the intellisense or documentation appear to be broken, for eg, the gradient link:
http://fabricjs.com/fabric-intro-part-2#gradients
returns a 404. Just wanted to flag this.

**Reply by asturur** [reply]
What part of the docs you are looking for? We lack extensive tutorial like docs. Are you struggling with something in particular?

**Reply by Itradetime** [reply]
Hi @asturur, not struggling now, I figured it out, but found it to be little confusing at first. When we visit fabricjs.com/docs, we see introduction to Fabric.js page, in there, if I click "Next" button, I am taken to old docs, for Fabric.js v5, not for the current major version. Instead of that, if we update it to "Getting started" page to be shown as home page, which has v6 topics, it would be better. 

It's not clear if "Api" section is for fabric v5 or v6, I assume it is v6, but it's not clear when getting started. Some sort of versioning at top would be more helpful. 

Also if I remember correctly, The docs sidebar has fabric v6 topics, then old docs, then API reference where versioning is not clear, and then a section called Fabric.js v5, which takes us to different webpage. This was confusing too. 

I agree one can figure it out if we look closer but it can be improved to be more consistent. Thanks for your time.

**Reply by asturur** [reply]
ohh that needs to be fixed yes.
The old docs folder shouldn't be there at all, i was supposed to convert each doc to something more modern more useful.
